packages/shell-api/src/explainable-cursor.ts

```diff
@@ -1,25 +1,29 @@
-import { shellApiClassDefault } from './decorators';
+import { shellApiClassDefault, returnType } from './decorators';
 import Cursor from './cursor';
 import Mongo from './mongo';
 import { asPrintable } from './enums';
-import { Cursor as ServiceProviderCursor } from '@mongosh/service-provider-core';
+import type { Document } from '@mongosh/service-provider-core';
 
 @shellApiClassDefault
 export default class ExplainableCursor extends Cursor {
-  _mongo: Mongo;
-  _cursor: ServiceProviderCursor;
+  _baseCursor: Cursor;
   _verbosity: string;
-  constructor(mongo, cursor, verbosity) {
-    super(mongo, cursor);
-    this._cursor = cursor;
-    this._mongo = mongo;
+
+  constructor(mongo: Mongo, cursor: Cursor, verbosity: string) {
+    super(mongo, cursor._cursor);
+    this._baseCursor = cursor;
     this._verbosity = verbosity;
   }
 
   /**
    * Internal method to determine what is printed for this class.
    */
   async [asPrintable](): Promise<any> {
-    return await this._cursor.explain(this._verbosity);
+    return await this._baseCursor.explain(this._verbosity);
+  }
+
+  @returnType('ExplainableCursor')
+  map(f: (doc: Document) => Document): ExplainableCursor {
+    return new ExplainableCursor(this._mongo, super.map(f), this._verbosity);
```

Comment:
Feel free to tell me if this doesn’t make sense this way :)

Reply:
I don't think the current driver explain would work with `map`, but generally with explain you're not interested in the result and you're rather supposed to just call `.explain` at the end. I'd wait on implementing this to see what  driver will do about it, there is a scope (already in progress ?) they are working on.